tools/scripts/generate_changelog.py

```diff
@@ -0,0 +1,67 @@
+"""
+Generate changelog for github release from the changelog.rst file
+Example of execution:
+
+python tools/scripts/generate_changelog.py -v 1.15.0 -f docs/changelog.rst
+
+It outputs the result to the stdout
+"""
+
+import argparse
+import re
+
+NEW_ASSETS_MSG = 'Added support for the following tokens'
+
+def generate_changelog(version: str, path: str) -> str:
+    change_regex = r'\*\ :(feature|bug):`(\d*|-)`\ (.+)'
+
+    changes = {'features': [], 'bugs': []}
+    github_link = '- #{number} {text}'
+    not_listed_change = '- {text}'
+
+    with open(path) as f:
+        lines = f.readlines()
+        start = False
+        for line in lines:
+            if version in line:
+                start = True
+                continue
+            if start:
+                r = re.search(change_regex, line)
+                if r is not None:
+                    groups = r.groups()
+                    # Check if the change is related to a commit
+                    if groups[1].isdigit():
+                        text = github_link.format(number=groups[1], text=groups[2])
+                    else:
+                        text = not_listed_change.format(text=groups[2])
+
+                    if NEW_ASSETS_MSG in groups[0]:
```

Comment:
This does not work still.

`python tools/scripts/generate_changelog.py -v 1.15.1 -f docs/changelog.rst` still generates the wrong changelog. You probably need to change the condition to:
```suggestion
                    if NEW_ASSETS_MSG in text:
```